lib/scripts/test-external.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+const fs = require('fs');
+const path = require('path');
+const { execSync } = require('child_process');
+const debug = require('debug')('test-external');
+
+const projectRoot = path.resolve(__dirname, '../../');
+const externalProjectName = process.argv[2];
+const gitUrl = process.argv[3];
+const tempDir = path.join(projectRoot, '../__external-test-cache');
```

Comment:
I intend to set this up such that we can cache the repositories we test on travis and use `git pull` to get latest most of the time vs needing to blow away and start over all the time.

Reply:
Alternatively, you could shallow clone for some speed improvement and not have to worry about caching mechanisms.

Reply:
@kategengler how does that work?

Reply:
`git clone --depth=1 <repo>` will get you just the HEAD of the repo, so you don't get all history.